README.md

```diff
@@ -4,4 +4,213 @@
   SPDX-License-Identifier: Apache-2.0
 -->
 
-# astarte-message-hub
+# Astarte Message Hub
+
+A central service that runs on (Linux) devices for collecting and delivering messages from N apps using 1 MQTT connection to Astarte.
+
+## Architecture
+The Astarte Message Hub is the main component that shares the Astarte connection to the nodes attached to it.
+The communication between the Hub and nodes is based on `gRPC` which is great for scenarios like real-time communication, low-power, low-bandwidth systems, and multi-language environments.
+`gRPC` makes the most out of `HTTP/2`, with multiplexed streaming and binary protocol framing. In addition, it offers performance advantages through the `Protobuf` message structure and features built-in code generation capability, which enables a multilingual environment.
+
+```mermaid
+flowchart LR
+  subgraph arch[  ]
+    direction TB
+    subgraph cloud[ ]
+        style cloud fill:#00b0f1,stroke:#ffffff
+        direction LR
+        subgraph astarte[Astarte]
+        style astarte fill:#127ab2,stroke:#ffffff,color:#fff
+        end
+    end
+    subgraph device[   ]
+    style device fill:#00b0f1, stroke:#ffffff
+    direction TB
+        astarte-mh[Astarte Message Hub]
+            style astarte-mh fill:#127ab2,stroke:#ffffff,color:#fff
+        astarte-mh <--> node1[Node 1]
+        astarte-mh <--> node2[Node 2]
+        astarte-mh <--> noden[Node...N]
+        style node1 stroke:#ffffff
+        style node2 stroke:#ffffff
+        style noden stroke:#ffffff
+    end
+  end
+  astarte <==> astarte-mh
+  style arch fill:#25516e,stroke:#c4c4c4,stroke-width:1 px,color:#fff,stroke-dasharray: 2 1
+```
+
+## GRPC MessageHub Service
+### Node
+A node is an entity connected to Astarte Message Hub, it can receive/send message from/to Astarte via the Message Hub.
+A node has a uniquely identified by its Node ID and has an Introspection that is a list of Interfaces used by it for exchanging data with Astarte. 
+
+``` protobuf
+message Node {
+  string id = 1;
```

Comment:
Are we enforcing some kind of structure for the id (e.g. UUID)? Maybe enforcing a UUID would help us in the future when we introduce the concept of Identities in Astarte, so we can map single nodes connected to a message hub to different Identities (that's mainly speculation at this point, but cc @bettio)